builtin/aws/apprunner/destroyer.go

```diff
@@ -0,0 +1,113 @@
+package apprunner
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apprunner"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/builtin/aws/utils"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// DestroyFunc implements component.Destroyer
+func (r *Releaser) DestroyFunc() interface{} {
+	return r.Destroy
+}
+
+func (r *Releaser) Destroy(
+	ctx context.Context,
+	log hclog.Logger,
+	ui terminal.UI,
+	release *Release,
+) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	step := sg.Add("Deleting Service")
+	// We put this in a function because if/when step is reassigned, we want to
+	// abort the new value.
+	defer func() {
+		step.Abort()
+	}()
+
+	// Delete app runner by service ARN
+
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Logger: log,
+	})
+	if err != nil {
+		step.Update(fmt.Sprintf("Failed to get session: %s", err))
+		return err
+	}
+
+	arSvc := apprunner.New(sess)
+
+	step.Update("Deleting service: %s", release.ServiceName)
+	dso, err := arSvc.DeleteService(&apprunner.DeleteServiceInput{
+		ServiceArn: aws.String(release.ServiceArn),
+	})
+	step.Done()
+
+	d := time.Now().Add(DEFAULT_TIMEOUT)
```

Comment:
10 minutes seems like a while! In testing, has it taken that long to delete the service?

Reply:
I've seem some operations take >7 minutes, so 10 minutes seemed like a reasonable value with _some buffer_ :|